Failed to build with OpenWrt toolchain

Сборка с `-DFAR2MACRO=no -DFAR2TVAR=no`
```
[395/396] Building CXX object far2l/CMakeFiles/far2l.dir/src/macro/nomacro.cpp.o
FAILED: far2l/CMakeFiles/far2l.dir/src/macro/nomacro.cpp.o 
/builder/staging_dir/host/bin/ccache /builder/staging_dir/toolchain-mipsel_24kc_gcc-11.2.0_musl/bin/mipsel-openwrt-linux-musl-g++ -DUNICODE -Dfar2l_EXPORTS -I/builder/build_dir/target-mipsel_24kc_musl/far2l-master/utils/include -I/builder/build_dir/target-mipsel_24kc_musl/far2l-master/far2l/. -I/builder/build_dir/target-mipsel_24kc_musl/far2l-master/far2l/far2sdk -I/builder/build_dir/target-mipsel_24kc_musl/far2l-master/far2l/src -I/builder/build_dir/target-mipsel_24kc_musl/far2l-master/far2l/src/base -I/builder/build_dir/target-mipsel_24kc_musl/far2l-master/far2l/src/mix -I/builder/build_dir/target-mipsel_24kc_musl/far2l-master/far2l/src/bookmarks -I/builder/build_dir/target-mipsel_24kc_musl/far2l-master/far2l/src/cfg -I/builder/build_dir/target-mipsel_24kc_musl/far2l-master/far2l/src/console -I/builder/build_dir/target-mipsel_24kc_musl/far2l-master/far2l/src/panels -I/builder/build_dir/target-mipsel_24kc_musl/far2l-master/far2l/src/filemask -I/builder/build_dir/target-mipsel_24kc_musl/far2l-master/far2l/src/hist -I/builder/build_dir/target-mipsel_24kc_musl/far2l-master/far2l/src/locale -I/builder/build_dir/target-mipsel_24kc_musl/far2l-master/far2l/src/macro -I/builder/build_dir/target-mipsel_24kc_musl/far2l-master/far2l/src/plug -I/builder/build_dir/target-mipsel_24kc_musl/far2l-master/far2l/src/vt -I/builder/build_dir/target-mipsel_24kc_musl/far2l-master/far2l -I/builder/build_dir/target-mipsel_24kc_musl/far2l-master/far2l/../WinPort -Os -pipe -mno-branch-likely -mips32r2 -mtune=24kc -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -msoft-float -mips16 -minterlink-mips16 -ffile-prefix-map=/builder/build_dir/target-mipsel_24kc_musl/far2l-master=far2l-master -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -D__MUSL__ -D__TAR_LIMITED_ARGS__ -Wall -fPIC -Wno-unused-function -D_FILE_OFFSET_BITS=64 -ffunction-sections -fdata-sections -Wl,--gc-sections -DWINPORT_REGISTRY -O2 -fvisibility=hidden -std=c++17 -MD -MT far2l/CMakeFiles/far2l.dir/src/macro/nomacro.cpp.o -MF far2l/CMakeFiles/far2l.dir/src/macro/nomacro.cpp.o.d -o far2l/CMakeFiles/far2l.dir/src/macro/nomacro.cpp.o -c /builder/build_dir/target-mipsel_24kc_musl/far2l-master/far2l/src/macro/nomacro.cpp
far2l/src/macro/nomacro.cpp:19:5: error: no declaration matches 'int KeyMacro::ProcessKey(FarKey)'
make[1]: *** [package/Makefile:123: package/feeds/far2l/far2l/compile] Error 1
   19 | int KeyMacro::ProcessKey(FarKey Key) { return 0; }
      |     ^~~~~~~~
In file included from far2l/src/macro/nomacro.cpp:15:
far2l/src/macro/macro.hpp:262:14: note: candidate is: 'bool KeyMacro::ProcessKey(FarKey)'
  262 |         bool ProcessKey(FarKey Key);
      |              ^~~~~~~~~~
far2l/src/macro/macro.hpp:191:7: note: 'class KeyMacro' defined here
  191 | class KeyMacro
      |       ^~~~~~~~
ninja: build stopped: subcommand failed.
make[2]: *** [Makefile:108: /builder/build_dir/target-mipsel_24kc_musl/far2l-master/.built] Error 1
make[2]: Leaving directory '/__w/far2l-openwrt/far2l-openwrt/far2l'
time: package/feeds/far2l/far2l/tiny/compile#5.02#7.44#5.01
    ERROR: package/feeds/far2l/far2l failed to build (build variant: tiny).
make[1]: Leaving directory '/builder'
make: *** [/builder/include/toplevel.mk:223: package/far2l/compile] Error 2
Error: Process completed with exit code 2.
```

[compile.zip](https://github.com/user-attachments/files/20853455/compile.zip)

--- a/far2l/bootstrap/scripts/FarEng.hlf.m4
+++ b/far2l/bootstrap/scripts/FarEng.hlf.m4
@@ -164,7 +164,7 @@ To work with these keys in FAR2L, you need to release keyboard shortcuts in the 
 
 
  #Special key mapping in macOS#
-    - keys #Option# map to #Alt#;
+    - The #Option# key maps to #Alt#;
     - far2l-GUI only: both keys #Command# act as Left #Ctrl#, both keys #Ctrl# act as Right #Ctrl#;
     - #Clear# key on the numeric keypad toggles NumLock mode;
     - numpad #0# functions as #Insert# when NumLock is off;

--- a/far2l/src/macro/nomacro.cpp
+++ b/far2l/src/macro/nomacro.cpp
@@ -16,7 +16,7 @@ nomacro.hpp
 
 KeyMacro::KeyMacro() {}
 KeyMacro::~KeyMacro() {}
-int KeyMacro::ProcessKey(FarKey Key) { return 0; }
+bool KeyMacro::ProcessKey(FarKey Key) { return false; }
 FarKey KeyMacro::GetKey() { return 0; }
 FarKey KeyMacro::PeekKey() { return 0; }
 bool KeyMacro::CheckWaitKeyFunc() { return false; }

--- a/far2l/src/message.cpp
+++ b/far2l/src/message.cpp
@@ -95,9 +95,8 @@ LONG_PTR WINAPI MsgDlgProc(HANDLE hDlg, int Msg, int Param1, LONG_PTR Param2)
 static int ShowMessageSynched(DWORD Flags, int Buttons, const wchar_t *Title, const wchar_t *const *Items,
 		int ItemsNumber, INT_PTR PluginNumber)
 {
-	FARString strTempStr;
 	int X1, Y1, X2, Y2;
-	int Length, BtnLength, J;
+	int Length, BtnLength;
 	DWORD I, MaxLength, StrCount;
 	BOOL ErrorSets = FALSE;
 	wchar_t *PtrStr;
@@ -231,14 +230,14 @@ static int ShowMessageSynched(DWORD Flags, int Buttons, const wchar_t *Title, co
 		ItemsNumber++;
 	}
 
-	for (J = 0; J < ItemsNumber - (EmptyText ? 1 : 0); ++J, ++I) {
+	for (int J = 0; J < ItemsNumber - (EmptyText ? 1 : 0); ++J, ++I) {
 		Str[I] = Items[J];
 	}
 
 	StrCount+= CountErrorLine;
 	MessageX1 = X1 = (ScrX - MaxLength) / 2 - 4;
 	MessageX2 = X2 = X1 + MaxLength + 9;
-	Y1 = (int(ScrY) - int(StrCount)) / 2 - 2;
+	Y1 = (int(ScrY) - int(StrCount) - 1) / 2 - 2;
 
 	if (Y1 < 0)
 		Y1 = 0;

--- a/far2l/src/panels/flmodes.cpp
+++ b/far2l/src/panels/flmodes.cpp
@@ -137,23 +137,22 @@ static const std::vector<PanelViewSettings> ViewSettingsRef = {
 
 std::vector<PanelViewSettings> ViewSettingsArray = ViewSettingsRef;
 
-inline static size_t DisplayModeToReal(size_t Mode)
+inline static int MenuPosToIndex(int MenuPos)
 {
-	return (Mode + 1) % 10;
+	return (MenuPos + 1) % 10;
 }
 
-inline static size_t RealModeToDisplay(size_t Mode)
+inline static int IndexToMenuPos(int Index)
 {
-	return (Mode + 9) % 10;
+	return (Index + 9) % 10;
 }
 
 void FileList::SetFilePanelModes()
 {
-	int CurMode = 0;
+	int MenuPos = 0;
 
 	if (CtrlObject->Cp()->ActivePanel->GetType() == FILE_PANEL) {
-		CurMode = CtrlObject->Cp()->ActivePanel->GetViewMode();
-		CurMode = RealModeToDisplay(CurMode);
+		MenuPos = IndexToMenuPos( CtrlObject->Cp()->ActivePanel->GetViewMode() );
 	}
 
 	for (;;) {
@@ -169,8 +168,7 @@ void FileList::SetFilePanelModes()
 			{Msg::EditPanelModesLinks,       0, 0},
 			{Msg::EditPanelModesAlternative, 0, 0}
 		};
-		int ModeNumber;
-		ModeListMenu[CurMode].SetSelect(1);
+		ModeListMenu[MenuPos].SetSelect(1);
 		{
 			VMenu ModeList(Msg::EditPanelModes, ModeListMenu, ARRAYSIZE(ModeListMenu), ScrY - 4);
 			ModeList.SetPosition(-1, -1, 0, 0);
@@ -188,7 +186,7 @@ void FileList::SetFilePanelModes()
 						if (Key & KEY_SHIFT) {
 							PanelPtr = CtrlObject->Cp()->GetAnotherPanel(PanelPtr);
 						}
-						PanelPtr->SetViewMode(DisplayModeToReal(ModeList.GetSelectPos()));
+						PanelPtr->SetViewMode( MenuPosToIndex(ModeList.GetSelectPos()) );
 						FrameManager->RefreshFrame(0); // side effect: changes macro area
 						CtrlObject->Macro.SetMode(MACROAREA_MENU); // restore macro area
 						break;
@@ -202,13 +200,11 @@ void FileList::SetFilePanelModes()
 				}
 			}
 
-			ModeNumber = ModeList.Modal::GetExitCode();
+			MenuPos = ModeList.Modal::GetExitCode();
 		}
 
-		if (ModeNumber < 0)
+		if (MenuPos < 0)
 			return;
-
-		CurMode = ModeNumber;
 
 		enum ModeItems
 		{
@@ -234,7 +230,7 @@ void FileList::SetFilePanelModes()
 			MD_BUTTON_CANCEL,
 		};
 		DialogDataEx ModeDlgData[] = {
-			{DI_DOUBLEBOX, 3,  1,  72, 15, {}, 0, ModeListMenu[ModeNumber].Name},
+			{DI_DOUBLEBOX, 3,  1,  72, 15, {}, 0, ModeListMenu[MenuPos].Name},
 			{DI_TEXT,      5,  2,  0,  2,  {}, 0, Msg::EditPanelModeTypes},
 			{DI_EDIT,      5,  3,  35, 3,  {}, DIF_FOCUS,                     L""},
 			{DI_TEXT,      5,  4,  0,  4,  {}, 0, Msg::EditPanelModeWidths},
@@ -259,9 +255,9 @@ void FileList::SetFilePanelModes()
 		int ExitCode;
 		RemoveHighlights(ModeDlg[MD_DOUBLEBOX].strData);
 
-		ModeNumber = DisplayModeToReal(ModeNumber);
+		int ModeIndex = MenuPosToIndex(MenuPos);
 
-		PanelViewSettings &NewSettings = ViewSettingsArray[ModeNumber];
+		PanelViewSettings &NewSettings = ViewSettingsArray[ModeIndex];
 		ModeDlg[MD_CHECKBOX_FULLSCREEN].Selected = NewSettings.FullScreen;
 		ModeDlg[MD_CHECKBOX_ALIGNFILEEXT].Selected = NewSettings.AlignExtensions;
 		ModeDlg[MD_CHECKBOX_ALIGNFOLDEREXT].Selected = NewSettings.FolderAlignExtensions;
@@ -282,7 +278,7 @@ void FileList::SetFilePanelModes()
 		}
 
 		if (ExitCode == MD_BUTTON_RESET) {
-			NewSettings = ViewSettingsRef[ModeNumber];
+			NewSettings = ViewSettingsRef[ModeIndex];
 		}
 		else if (ExitCode == MD_BUTTON_OK) {
 			memset(&NewSettings, 0, sizeof(NewSettings));
@@ -307,8 +303,8 @@ void FileList::SetFilePanelModes()
 		CtrlObject->Cp()->SetScreenPosition();
 		int LeftMode = CtrlObject->Cp()->LeftPanel->GetViewMode();
 		int RightMode = CtrlObject->Cp()->RightPanel->GetViewMode();
-		//	CtrlObject->Cp()->LeftPanel->SetViewMode(ModeNumber);
-		//	CtrlObject->Cp()->RightPanel->SetViewMode(ModeNumber);
+		//	CtrlObject->Cp()->LeftPanel->SetViewMode(ModeIndex);
+		//	CtrlObject->Cp()->RightPanel->SetViewMode(ModeIndex);
 		CtrlObject->Cp()->LeftPanel->SetViewMode(LeftMode);
 		CtrlObject->Cp()->RightPanel->SetViewMode(RightMode);
 		CtrlObject->Cp()->LeftPanel->Redraw();
@@ -318,8 +314,8 @@ void FileList::SetFilePanelModes()
 
 void FileList::ReadPanelModes(ConfigReader &cfg_reader)
 {
-	for (int I = 0; I < ViewSettingsArray.size(); I++) {
-		cfg_reader.SelectSectionFmt("Panel/ViewModes/Mode%d", I);
+	for (size_t I = 0; I < ViewSettingsArray.size(); I++) {
+		cfg_reader.SelectSectionFmt("Panel/ViewModes/Mode%d", (int)I);
 		FARString strColumnTitles = cfg_reader.GetString("Columns", L"");
 		FARString strColumnWidths = cfg_reader.GetString("ColumnWidths", L"");
 		FARString strStatusColumnTitles = cfg_reader.GetString("StatusColumns", L"");
@@ -351,8 +347,8 @@ void FileList::ReadPanelModes(ConfigReader &cfg_reader)
 
 void FileList::SavePanelModes(ConfigWriter &cfg_writer)
 {
-	for (int I = 0; I < ViewSettingsArray.size(); I++) {
-		cfg_writer.SelectSectionFmt("Panel/ViewModes/Mode%d", I);
+	for (size_t I = 0; I < ViewSettingsArray.size(); I++) {
+		cfg_writer.SelectSectionFmt("Panel/ViewModes/Mode%d", (int)I);
 
 		FARString strColumnTitles, strColumnWidths, strStatusColumnTitles, strStatusColumnWidths;
 

